shiny-ci-cd.Rmd

```diff
@@ -34,9 +34,72 @@ Generally speaking, a CI/CD pipeline related to an R package is comprised of the
+[Workflows](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions) use `YAML` syntax and should be stored in the `.github/workflows` directory in the root of the repository.
+Workflows are constituted of jobs and each job is a set of steps to perform individual tasks, e.g. commands or actions.
+
+### Workflow steps
+
+A workflow should have an identifying `name` and an `on` section that indicates upon which events the workflow should be triggered. It should include at least one job and each job will have a set of steps fully specifying what to execute. Such steps can be an action (predefined, sourcing from GitHub repos that contain such actions) or a script. However, for the time being, GitHub Actions does not support a hierarchical, aggregated structure of actions. Being entirely customizable, it is necessary to fully specify each step in the CI/CD pipeline explicitly.
+
+#### Setup
+
+- Checkout the source package from the current repo, using `actions/checkout` provided by GitHub.
+- Setup R using the action [`r-lib/actions/setup-r`](https://github.com/r-lib/actions#readme).
+- Query and cache R package dependencies using `remotes::dev_package_deps()` and the `actions/cache` predefined action.
+- Install system dependencies using package `sysreqs::sysreq_commands()` (for the `ubuntu` runner used to run the action).
+- Install R package dependencies using `remotes::Install_deps()`.
+
+#### Package check
+
+- Check the package via `rcmdcheck::rcmdcheck()`.
+
+#### Deployment
+
+- Deploy to shinyapps.io:
+  - In this case, environment variables `SHINYAPPS_ACCOUNT`, `SHINYAPPS_TOKEN` and `SHINYAPPS_SECRET` (defining credentials for [shinyapps.io](https://www.shinyapps.io/)), are specified / accessible as GitHub [secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference#authentication-and-secrets).
+  - An R script e.g. `deploy/deploy-shinyapps.R` (build-ignored via `usethis::use_build_ignore("deploy")`) defines the deployment commands based on the environment variables:
+```{r read-deploy-script, eval = TRUE, echo = FALSE}
+knitr::read_chunk("shiny-ci-cd/deploy/deploy-shinyapps.R", labels = "deploy-shinyapps")
+```
+```{r deploy-shinyapps}
+```
+
+### Workflow file
+
+The `steps` described in the previous section are defined in the `.yml` workflow file as follows:
+
+```{r read-ci-cd.yml, eval = TRUE, include = FALSE}
+file.copy("shiny-ci-cd", "_book", recursive = TRUE)
+knitr::read_chunk("shiny-ci-cd/actions/ci-cd.yml", labels = "ci-cd.yml")
+```
+```{yml ci-cd.yml}
+```
+
+As visible from the run logs that can be found on GitHub under the `Actions` tab, all the CI/CD pipeline steps are performed subsequently, and are identifiable by the `name` field. See the example below:
+
+![GitHub Actions Continuous Integration / Continuous Deployment pipeline for a packaged Shiny app](shiny-ci-cd/img/ShinyCICD_githubactions1.png)
```

Comment:
We need a new image now to be aligned with the steps as revised

Reply:
I could take a screenshot from https://github.com/miraisolutions/ShinyCICD/runs/2508242565. For some reason, the title changed from CI-CD / CI-CD to ubuntu-20.04 (release). Let me know if we should investigate this or just assume that something changed in the background on how this title is derived and that it is fine to keep the new name.

Reply:
I agree that a screenshot from the run you propose will do.

> the title changed from CI-CD / CI-CD to ubuntu-20.04 (release)

This is due to the fact that we set the `job`  `name: ` to the matrix entries for OS and R version. This makes sense and I think it is OK. CI-CD is also the name of the workflow and is now shown in the left side bar and not in the title of the run  output, which now only contains the job name.

Reply:
updated and pushed